File: helpdesk-pro/frontend/src/app/core/services/task.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { environment } from '../../../environments/environment';
import { ApiResponse } from '../models';

export interface TicketTask {
  id: string;
  ticketId: string;
  title: string;
  completed: boolean;
  assignedToId: string | null;
  assignedToName: string | null;
  dueDate: string | null;
  createdAt: string;
}

@Injectable({ providedIn: 'root' })
export class TaskService {
  private base = `${environment.apiUrl}/tickets`;

  constructor(private http: HttpClient) {}

  getTasks(ticketId: string): Observable<TicketTask[]> {
    return this.http.get<ApiResponse<TicketTask[]>>(`${this.base}/${ticketId}/tasks`).pipe(map(r => r.data));
  }

  createTask(ticketId: string, title: string): Observable<TicketTask> {
    return this.http.post<ApiResponse<TicketTask>>(`${this.base}/${ticketId}/tasks`, { title }).pipe(map(r => r.data));
  }

  toggleTask(ticketId: string, taskId: string): Observable<TicketTask> {
    return this.http.patch<ApiResponse<TicketTask>>(`${this.base}/${ticketId}/tasks/${taskId}`, {}).pipe(map(r => r.data));
  }

  deleteTask(ticketId: string, taskId: string): Observable<void> {
    return this.http.delete<void>(`${this.base}/${ticketId}/tasks/${taskId}`);
  }
}

File: helpdesk-mobile/lib/core/services/task_service.dart
import '../api/api_client.dart';
import '../api/api_endpoints.dart';
import '../models/task_model.dart';

class TaskService {
  final ApiClient _api;
  TaskService(this._api);

  Future<List<TaskModel>> getTasks(String ticketId) async {
    final resp = await _api.get(ApiEndpoints.ticketTasks(ticketId));
    final data = resp.data['data'];
    final list = data is List ? data : (data['content'] ?? []);
    return List<TaskModel>.from(list.map((e) => TaskModel.fromJson(e)));
  }

  Future<TaskModel> createTask(String ticketId, String title) async {
    final resp = await _api.post(ApiEndpoints.ticketTasks(ticketId), data: {'title': title});
    return TaskModel.fromJson(resp.data['data']);
  }

  Future<TaskModel> toggleTask(String ticketId, String taskId) async {
    final resp = await _api.post(ApiEndpoints.ticketTaskToggle(ticketId, taskId));
    return TaskModel.fromJson(resp.data['data']);
  }

  Future<void> deleteTask(String ticketId, String taskId) async {
    await _api.delete(ApiEndpoints.ticketTask(ticketId, taskId));
  }
}
